--- origsrc/xosd-2.2.14/src/bmp_plugin/Makefile.am	2004-11-08 14:34:39.000000000 -0600
+++ src/xosd-2.2.14/src/bmp_plugin/Makefile.am	2007-04-08 02:14:57.578125000 -0500
@@ -19,8 +19,8 @@
 bmpplugin_LTLIBRARIES = $(NEW_bmpplugin) $(OLD_bmpplugin)
 
 libbmp_osd_la_SOURCES = bmp_osd.c dlg_config.c dlg_font.c dlg_colour.c bmp_osd.h
-libbmp_osd_la_LIBADD  = $(top_builddir)/src/libxosd/libxosd.la
-libbmp_osd_la_LDFLAGS = -module -avoid-version @GDK_PIXBUF_LIBS@
+libbmp_osd_la_LIBADD  = $(top_builddir)/src/libxosd/libxosd.la $(BMP_LIBS)
+libbmp_osd_la_LDFLAGS = -module -avoid-version -no-undefined
 
 libbmp_osd_old_la_SOURCES = bmp_osd.c dlg_config_old.c dlg_font.c dlg_colour.c bmp_osd.h
 libbmp_osd_old_la_LIBADD  = $(top_builddir)/src/libxosd/libxosd.la
--- origsrc/xosd-2.2.14/src/libxosd/Makefile.am	2004-08-19 17:39:33.000000000 -0500
+++ src/xosd-2.2.14/src/libxosd/Makefile.am	2007-04-08 02:10:44.343750000 -0500
@@ -3,6 +3,6 @@
 lib_LTLIBRARIES 	= libxosd.la
 libxosd_la_SOURCES 	= xosd.c intern.h
 libxosd_la_LIBADD 	= $(X_LIBS)
-libxosd_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -pthread
+libxosd_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 
  
--- origsrc/xosd-2.2.14/src/xmms_plugin/Makefile.am	2004-08-19 17:39:31.000000000 -0500
+++ src/xosd-2.2.14/src/xmms_plugin/Makefile.am	2007-04-08 02:23:28.812500000 -0500
@@ -17,8 +17,8 @@
 xmmsplugin_LTLIBRARIES = $(NEW_xmmsplugin) $(OLD_xmmsplugin)
 
 libxmms_osd_la_SOURCES = xmms_osd.c dlg_config.c dlg_font.c dlg_colour.c xmms_osd.h
-libxmms_osd_la_LIBADD  = $(top_builddir)/src/libxosd/libxosd.la
-libxmms_osd_la_LDFLAGS = -module -avoid-version @GDK_PIXBUF_LIBS@
+libxmms_osd_la_LIBADD  = $(top_builddir)/src/libxosd/libxosd.la $(XMMS_LIBS) $(GDK_PIXBUF_LIBS)
+libxmms_osd_la_LDFLAGS = -module -avoid-version -no-undefined
  
 libxmms_osd_old_la_SOURCES = xmms_osd.c dlg_config_old.c dlg_font.c dlg_colour.c xmms_osd.h
 libxmms_osd_old_la_LIBADD  = $(top_builddir)/src/libxosd/libxosd.la
